File: backend/app/schemas/review_comments.py
from pydantic import BaseModel, Field


class ReviewCommentCreateRequest(BaseModel):
    comment_text: str = Field(min_length=1)
    created_by_user_id: str = Field(min_length=1)


class ReviewCommentResponse(BaseModel):
    comment_id: int
    review_item_id: int
    comment_text: str
    created_by_user_id: str
    created_utc: int


class ReviewCommentListResponse(BaseModel):
    items: list[ReviewCommentResponse]

File: frontend/src/services/reviewComments.ts
import { apiGet, apiPost } from "./api";

export type ReviewComment = {
  comment_id: number;
  review_item_id: number;
  comment_text: string;
  created_by_user_id: string;
  created_utc: number;
};

export type ReviewCommentListResponse = {
  items: ReviewComment[];
};

export type ReviewCommentCreateRequest = {
  comment_text: string;
  created_by_user_id: string;
};

export function getReviewComments(reviewItemId: number) {
  return apiGet<ReviewCommentListResponse>(`/review-items/${reviewItemId}/comments`);
}

export function createReviewComment(
  reviewItemId: number,
  payload: ReviewCommentCreateRequest,
) {
  return apiPost<ReviewComment, ReviewCommentCreateRequest>(
    `/review-items/${reviewItemId}/comments`,
    payload,
  );
}